Source/LanderLads/SpaceshipPawn.cpp

```diff
@@ -0,0 +1,226 @@
+}
+
+void ASpaceshipPawn::MoveLeftButtonReleased()
+{
+	this->LeftThrustStartTime = 0.0f;
+}
+
+void ASpaceshipPawn::MoveForwardButtonPressed()
+{
+	this->ForwardThrustStartTime = this->GetWorld()->GetRealTimeSeconds();
+}
+
+void ASpaceshipPawn::MoveForwardButtonReleased()
+{
+	this->ForwardThrustStartTime = 0.0f;
+}
+
+void ASpaceshipPawn::MoveBackwardButtonPressed()
+{
+	this->BackwardThrustStartTime = this->GetWorld()->GetRealTimeSeconds();
+}
+
+void ASpaceshipPawn::MoveBackwardButtonReleased()
+{
+	this->BackwardThrustStartTime = 0.0f;
+}
+
+
+void ASpaceshipPawn::MoveForwardBackward(float AxisValue)
+{
+	constexpr float RelativeXThrust = 490 * 4000;
+	const float ClampedAxisValue = ClampAxisValue(AxisValue);
+
+	const float TimeDifference = ClampedAxisValue > 0.0f ?
+		(this->GetWorld()->GetRealTimeSeconds() - this->ForwardThrustStartTime) :
+		(this->GetWorld()->GetRealTimeSeconds() - this->BackwardThrustStartTime);
+	
+	const float Curvey = this->CurveFloat->GetFloatValue(TimeDifference);
+	const float ForwardBackwardVectorMagnitude = RelativeXThrust * ClampedAxisValue * Curvey;
+	this->ResultantForceVector.SetComponentForAxis(EAxis::X, ForwardBackwardVectorMagnitude);
+}
+
+void ASpaceshipPawn::MoveLeftRight(float AxisValue)
+{
+	constexpr float RelativeYThrust = 490 * 4000;
+	const float ClampedAxisValue = ClampAxisValue(AxisValue);
+
+	const float TimeDifference = ClampedAxisValue > 0.0f ?
+		(this->GetWorld()->GetRealTimeSeconds() - this->RightThrustStartTime) :
+		(this->GetWorld()->GetRealTimeSeconds() - this->LeftThrustStartTime);
+
+	const float Curvey = this->CurveFloat->GetFloatValue(TimeDifference);
+	const float ForwardBackwardVectorMagnitude = RelativeYThrust * ClampedAxisValue * Curvey;
+	this->ResultantForceVector.SetComponentForAxis(EAxis::Y, ForwardBackwardVectorMagnitude);
+}
+
+void ASpaceshipPawn::MoveUpDown(float AxisValue)
+{
+	const float Lift = this->GetWorld()->GetGravityZ() * -1 * 4000;
+	const float ClampedAxisValue = ClampAxisValue(AxisValue);
+
+	const float TimeDifference = ClampedAxisValue > 0.0f ? 
+		(this->GetWorld()->GetRealTimeSeconds() - this->UpThrustStartTime) : 
+		(this->GetWorld()->GetRealTimeSeconds() - this->DownThrustStartTime);
+
+	const float Curvey = this->CurveFloat->GetFloatValue(TimeDifference);
+	const float UpDownForceVectorMagnitude = Lift * ClampedAxisValue * Curvey;
+	this->ResultantForceVector.SetComponentForAxis(EAxis::Z, UpDownForceVectorMagnitude);
+	
+}
+
+void ASpaceshipPawn::MoveCameraLookHorizontal(float AxisValue)
+{
+	float ClampedAxisValue = ClampAxisValue(AxisValue);
+	this->AddControllerYawInput(
+		this->GetWorld()->GetDeltaSeconds() * 45 * ClampedAxisValue);
+}
+
+void ASpaceshipPawn::MoveCameraLookVertical(float AxisValue)
+{
+	float ClampedAxisValue = ClampAxisValue(AxisValue);
+	this->AddControllerPitchInput(
+		this->GetWorld()->GetDeltaSeconds() * -45 * ClampedAxisValue);
+}
+
+const float ASpaceshipPawn::ClampAxisValue(float AxisValue)
```

Comment:
Dangerous as it hides magic values inside a one-liner.  If the clamp must be [-1.0f, 1.0f] then justify it in the context.